### ISSUE TITLE
Provide a new macro only updating current label

### ISSUE THREAD

**Issue by muzimuzhi** [latex3/latex2e#708]
## Brief outline of the enhancement

Package authors sometimes want to control the value of a counter while make its info still stored as current label. Currently they have to use
```tex
\addtocontents{mycnt}{-1}%
\refstepcounter{mycnt}%
```
to solely update the current label (`\@currentcounter` and `\@currentlabel`). With the new macro, `\refstepcounter` can then be defined as simple as a combination of two more atomic macros.

**Comment by u-fischer** [comment]
It is not quite clear imho what exactly the command should do. E.g. if hyperref is loaded there is also the anchor and the title for nameref to consider.

**Comment by muzimuzhi** [comment]
My vague thinking is, it (let's call it `\@refcounter` just for convenience) does all things `\refstepcounter{<cnt>}` do, except for `\stepcounter{<cnt>}`. `hyperref` can then modify `\@refcounter` to insert new things.

Well it's maybe too late (for the kernel) to introduce new macros and make them widely used...

**Comment by u-fischer** [comment]
> Well it's maybe too late (for the kernel) to introduce new macros and make them widely used...

No, we will have to rework this area anyway. I only meant that one will have to consider hyperref directly. I don't want a command that hyperref then has to patch.
